Test Finish

ThomasPetithory opened the pull request:
Manque encore la bonus task mais ce sera fait d'ici la semaine prochaine

Commit:
Test Finish

Commit:
Bonus Task Finish

Commit:
Correction pour l'env prod

Commit:
Rajout de qq commentaires, nettoyage de résiduts de tests, etc

benjamin-hubert reviewed (changes requested):
Thomas,

Tu as commit tes .idea, qui sont propres à ton IDE de développement (PHPStorm du coup).

Ce n'est pas une bonne pratique car ça écraserait nos configs si on récupérait le projet. J'aimerai bien que tu rajoutes cet élément dans le git ignore et que tu fasses en sorte qu'ils disparaissent de ta PR.

Je te laisse regarder ;).

Commit:
delete .idea

benjamin-hubert commented on app/Resources/views/default/index.html.twig:
```diff
@@ -39,6 +41,14 @@
+                    <br/>
+                    <!-- petit rajout pour avoir les liens directement sur l'acceuil -->
+                    <ul>
+                        <li><a href={{ path('test_interview_homepage') }}>Index du bundle Test</a></li>
```
Attention à tes liens :)

ThomasPetithory replied:
Pourquoi? Est-ce que je dois rajouter une vérification que la route existe?

benjamin-hubert replied:
Non, il manque des guillemets tout simplement dans ton href ;)

ThomasPetithory replied:
Oupsss.... Corrigé en même temps que swiftmailer

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
 
 1. Run the PhpUnit test. Check if there are any errors, if so fix them.
+OK (en mettant en commentaires "swiftmailer" dans config_test.yml et modifier le test)
+(commande à la racine : "phpunit -c app/ src/AppBundle/Tests/Controller/DefaultControllerTest.php")
```
Je ne comprend pas pourquoi tu as du désactiver le swiftmailer ? Qu'est-ce qui se passait ?

benjamin-hubert replied:
OK j'ai vu en voulant faire le test, essaies de résoudre le problème sans virer ça du config_test :)

ThomasPetithory replied:
Ok j'ai rajouter un require pour le swiftmailer bundle et fais un composer install, et là c'est ok, je l'ai pas configuré par contre

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
 1. Create a method helloAction under AppBundle\Controller\DefaultController
   * for route `/hello`
   * with a proper json return `{"hello":"world!"}`
+Ok (return JsonResponse)
```
👍

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
+(commande à la racine : "phpunit -c app/ src/AppBundle/Tests/Controller/DefaultControllerTest.php")
 
 1. Create a new Bundle "InterviewBundle" within the namespace "Test"
+OK (changement de l'index de l'appli avec la commande generate:bundle)
```
👍

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
+(commande à la racine pour les setters/getters: "php app/console doctrine:mongodb:generate:documents TestInterviewBundle")
 
 1. Define ODM "Bios" repository under namespace Test/InterviewBundle/Repositories
+Ok sauf qu'on doit utiliser "Repository" et non "Repositories"
```
Effectivement la bonne pratique est Repository mais ça aurait pu fonctionner aussi avec Repositories ;)

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
 
 1. Define ODM "Bios" document under namespace Test/InterviewBundle/Documents
+Ok sauf qu'on ne doit pas utiliser "Documents" mais "Document" pour que le mapping fonctionne (pas trouvé d'autres solutions)
+(commande à la racine pour les setters/getters: "php app/console doctrine:mongodb:generate:documents TestInterviewBundle")
```
👍

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
   * Use the logger to log operations (error, warning, debug)
+Ok, inscrit dans les services du Bundle (pour la portabilité)
+Cependant, j'ai utilisé ma classe test pour tester le service, mais je ne me suis pas servi du logger
 
```
Ca serait bien d'implémenter le logger stp :). 

Sur les retours, un service ne doit pas renvoyer de Response (une Response c'est essentiellement un objet qui HTTP pour renvoeyr des données au navigateur et donc, envoyé que par le controller). Est-ce que tu peux modifier un peu le code pour que ça soit plus propre là dessus ? 

De même, ton renvoi "false" depuis tes Repository me semble un peu brouillon car je pense qu'il y'a moyen d'éviter ça derrière, mais je termine la revue pour te le confirmer ! 

Sinon le reste me semble bien !

ThomasPetithory replied:
J'ai injecter le logger dans le BiosService

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
   * method should list all contributions
   * with a proper json return `["contrib", ...]`
+OK
 
```
Bon, ça va venir contredire mon post précédent de "un service ne doit pas renvoyer de Response" car dans ce cas précis, je comprend pourquoi tu l'as fait.

Je pense que je n'aurai pas géré ça comme ça (genre je pense que j'aurai assigné le résultat du service dans une variable et renvoyé la response depuis le controller pour des questions de réutilisation du code). Mais ce que tu as fait fonctionne et on est sur de l'affinage donc pas de souci !

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
   * method should list all bios documents with provided contribution
   * with a proper json return `[{...}]`
+Ok
 
```
idem

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
-  
+Ok j'ai mis les tests dans Test/InterviewBundle/Tests/Controller/DefaultControllerTest.php
+(commande à la racine: "phpunit -c app/ src/Test/InterviewBundle//Tests/Controller/DefaultControllerTest.php") 
+ 
```
👍

benjamin-hubert commented on README.md:
```diff
@@ -42,65 +42,92 @@ Access the project with your favourite browser. You should see similar welcome s
 1. make a unit test for the BiosService
   * at least 1 method of your choice
+Ok j'ai mis un test de 'getAllContributions' avec retour Json à la suite des tests précédents
 
```
Alors, si tu test le service, il faut créer une nouvelle class (et pas mettre ton test dans la classe de test du controller).

Le test est pas mal sinon !

ThomasPetithory replied:
J'ai séparé les deux tests et corriger celui de service : 
en fait, il vérifiait la réponse de $client->request('GET', '/contributions/OOP'); (qui rend un json) situé au dessus et non le retour du service!

benjamin-hubert commented on README.md:
```diff
@@ -109,14 +136,17 @@ Access the project with your favourite browser. You should see similar welcome s
 1. Check the symfony application for errors and fix them if any.
+Ok, j'ai du modifier "test:" en "test_interview:" qui est le namespace du bundle
+et ajouter du code dans le DependencyInjection/Configuration pour ajouter le paramètre ping de valeur pong (valeur par défaut obligatoire)
 
```
👍

benjamin-hubert commented on README.md:
```diff
@@ -109,14 +136,17 @@ Access the project with your favourite browser. You should see similar welcome s
   * If you accept, run the command
-
+OK, j'ai du modifier le code demandé en 15 forcément
 
```
Ca marche, avec un check sur le paramètre qui devrait être obligatoire serait encore mieux ;)

ThomasPetithory replied:
Well done, mais c'est juste une vérification d'existence, un message et un exit

Commit:
rajout du gitignor
correction des guillemets pour les liens
ajout d'un composer require pour le swiftmailer bundle

Commit:
Séparation des Tests Controller et Service
Correction du test du service qui ne vérifiait pas la bonne réponse

Commit:
Ajout d'une vérification d'existence de parametre sur la commande et message d'erreur sinon

Commit:
Injection du logger dans le BiosService
